src/main/java/io/lettuce/core/dynamic/SimpleBatcher.java

```diff
@@ -166,7 +166,6 @@ private List<RedisCommand<Object, Object, Object>> prepareDefaultFlush(int consu
 
             RedisCommand<Object, Object, Object> poll = queue.poll();
 
-            assert poll != null;
             batch.add(poll);
```

Comment:
The code here in `prepareDefaultFlush()` is already checking if the queue is empty before poll()ing from it, so I didn't have to do anything.